spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/PollableConsumerBinder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder;
+
+/**
+ * A binder that supports pollable message sources.
+ *
+ * @param <H> the polled consumer handler type.
+ * @param <C> the consumer properties type.
+ * @param <P> the producer properties type.
+ *
+ * @author Gary Russell
+ * @since 2.0
+ *
+ */
+public interface PollableConsumerBinder<H, C extends ConsumerProperties> {
+
+	/**
+	 * Configure a binding for a pollable message source.
+	 * @param name the binding name.
+	 * @param group the consumer group.
+	 * @param inboundBindTarget the binding target.
+	 * @param consumerProperties the consumer properties.
+	 * @return the binding.
+	 */
+	default Binding<PollableSource<H>> bindPollableConsumer(String name, String group,
```

Comment:
I guess I need to see how it is used in the actual binder implementation, but at least for now I am wondering if it would be better NOT to provide a default implementation and instead force user to either implement the operation otherwise it is not `PollableConsumerBinder`.

Reply:
Yes; this is a holdover from when I had this as a subinterface of Binder; the AMCB has a default that protects other binders (JMS, Kinesis).

I can create the PR for Rabbit if you like, but the PR won't pass travis until this is merged.

Reply:
That's fine, let's merge this one first